### PR TITLE
smallrye-fault-tolerance-9 MP FT Spec 1.1 updates

### PR DESCRIPTION
**Pull request by antoinesd**
Added global config support

### CODE REVIEW

File: implementation/src/main/java/io/smallrye/faulttolerance/FaultToleranceMetricsPublisher.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import com.netflix.hystrix.strategy.metrics.HystrixMetricsPublisherCollapser;
+import com.netflix.hystrix.strategy.metrics.HystrixMetricsPublisherCommand;
+import com.netflix.hystrix.strategy.metrics.HystrixMetricsPublisherThreadPool;
+
+public class FaultToleranceMetricsPublisher extends HystrixMetricsPublisher {
+
+
+    @Override
+    public HystrixMetricsPublisherCommand getMetricsPublisherForCommand(HystrixCommandKey commandKey, HystrixCommandGroupKey commandGroupKey, HystrixCommandMetrics metrics, HystrixCircuitBreaker circuitBreaker, HystrixCommandProperties properties) {
+        System.out.println("**** Here ****");
```
Comment (mkouba) [inline implementation/src/main/java/io/smallrye/faulttolerance/FaultToleranceMetricsPublisher.java]: Remove or maybe some more meaningful message? Also we use `jboss-logging` in `smallrye-fault-tolerance`...

File: implementation/src/main/java/io/smallrye/faulttolerance/FaultToleranceCommandExecutionHook.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+public class FaultToleranceCommandExecutionHook extends HystrixCommandExecutionHook {
+
+
+    @Override
+    public <T> Exception onExecutionError(HystrixInvokable<T> commandInstance, Exception e) {
+
+        SimpleCommand command = ((SimpleCommand) commandInstance);
+
+        command.setFailure(e);
+
```
Comment (mkouba) [inline implementation/src/main/java/io/smallrye/faulttolerance/FaultToleranceCommandExecutionHook.java]: Too many blank lines ;-)

File: implementation/src/main/java/io/smallrye/faulttolerance/HystrixCommandInterceptor.java [hunk 3]
```diff
@@ -190,18 +212,107 @@ private Object executeCommand(Function<Supplier<Object>, SimpleCommand> commandF
...
+                if(command.isResponseFromFallback()) {
+                    counterInc(prefix + ".fallback.calls.total");
+                }
+
             }
         }
     }
 
+    private void counterInc(String name) {
+        //TODO:disable metrics by config
```
Comment (mkouba) [inline implementation/src/main/java/io/smallrye/faulttolerance/HystrixCommandInterceptor.java]: I think this is essential... not all users are interested in metrics.

**Comment by antoinesd** [comment]
Optional approach is now integrated

**Comment by mkouba** [comment]
@antoinesd pls rebase...

File: implementation/src/main/java/io/smallrye/faulttolerance/FaultToleranceMetricsPublisher.java (hunk 1 above)
Comment (antoinesd) [reply implementation/src/main/java/io/smallrye/faulttolerance/FaultToleranceMetricsPublisher.java]: File doesn't exist anymore

File: implementation/src/main/java/io/smallrye/faulttolerance/HystrixCommandInterceptor.java (hunk 3 above)
Comment (antoinesd) [reply implementation/src/main/java/io/smallrye/faulttolerance/HystrixCommandInterceptor.java]: It is now integrated

File: implementation/src/main/java/io/smallrye/faulttolerance/FaultToleranceCommandExecutionHook.java (hunk 2 above)
Comment (antoinesd) [reply implementation/src/main/java/io/smallrye/faulttolerance/FaultToleranceCommandExecutionHook.java]: done

**Comment by antoinesd** [comment]
rebase done

**Pull request merged**